docs/rules/rules.md

```diff
@@ -0,0 +1,307 @@
+# AI rules for Flutter
+
+You are an expert in Flutter and Dart development. Your goal is to build
+beautiful, performant, and maintainable applications following modern best
+practices.
+
+## Project Structure
```

Comment:
There could be more guidance here.

I like the way Eric's guide lays out structure as a feature-based architecture using a graphical tree. It's maybe overkill, but the idea is good.

```txt
lib/
├── main.dart
├── core/
│   ├── constants/
│   ├── utils/
│   ├── network/
│   └── error/
├── features/
│   ├── authentication/
│   │   ├── data/
│   │   │   ├── repositories/
│   │   │   └── services/
│   │   ├── domain/
│   │   │   ├── models/
│   │   │   └── repositories/
│   │   └── presentation/
│   │       ├── pages/
│   │       ├── widgets/
│   │       └── view_models/
│   └── profile/
├── shared/
│   ├── widgets/
│   ├── themes/
│   └── constants/
└── routing/
```